salt/_pillar/metalk8s.py

```diff
@@ -15,8 +15,17 @@ def _load_config(path):
     log.debug('Loading MetalK8s configuration from %s', path)
 
     config = None
-    with salt.utils.files.fopen(path, 'rb') as fd:
-        config = salt.utils.yaml.safe_load(fd)
+    try:
+        with salt.utils.files.fopen(path, 'rb') as fd:
+            config = salt.utils.yaml.safe_load(fd) or {}
+    except Exception as exc:
+        return __utils__['pillar_utils.errors_to_dict']([
+            "Failed to load {}: {}".format(path, exc)
+        ])
+    if not config or config is None:
+        return __utils__['pillar_utils.errors_to_dict']([
+            "Unable to load an empty BootstrapConfiguration at {}".format(path)
```

Comment:
Message could be instead: `Invalid BootstrapConfiguration at {path}: must not be empty.`